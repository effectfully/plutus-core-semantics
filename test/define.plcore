(program

(module Foo (imported) (exported () ()) (local () ())
  (define four (builtin addInt 2 2))

  (define result (builtin multiplyInt Foo.four Foo.four))
)

)
