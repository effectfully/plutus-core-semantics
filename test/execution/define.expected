<T>
  <k> . </k>
  <lenv> .Map </lenv>
  <genv> ( Foo . result ) |-> 16 ( Foo . four ) |-> 4 </genv>
  <modules>
    <Module>
      <curr> 0 </curr>
      <name> Foo </name>
      <dataCons> .TyExpList </dataCons>
      <decl> four  .NameList </decl>
      <importedModules> .ModNmList </importedModules>
      <exportTy> .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </Module>
  </modules>
</T>
